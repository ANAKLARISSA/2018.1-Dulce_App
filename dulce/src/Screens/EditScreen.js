import React from 'react';
import {View, TextInput, Text} from 'react-native';
import AGRButton from '../Components/AGRButton';


const styles = {
  container:{
    flex: 1,
    flexDirection: 'column',
    padding: 15,
    backgroundColor: '#FFF',
    paddingHorizontal: 20,
  },
  input:{
    height: 36,
    borderBottomWidth: 1,
    borderBottomColor: 'grey',
    marginBottom: 10,
    alignSelf: 'center',
    width: '80%',
    marginTop: 10,
  },
  button: {
    alignItems: 'center',
    backgroundColor: 'pink',
    padding: 10,
    color: 'blue'
  },
};

class EditScreen extends React.Component {
  constructor(props){
    super(props);
    this.state = {editable:false};
  }

  tornarVisivel(){
    this.setState({
      editable: !this.state.editable
    });
  }


  render() {
    return (
      <View style={styles.container}>
        <TextInput style={styles.input}
          placeholder='Nome'
          editable = {this.state.editable}
        />
        <TextInput style={styles.input}
          placeholder='Matricula'
          editable = {this.state.editable}
        />
        <TextInput style={styles.input}
          placeholder='Hospital'
          editable = {this.state.editable}
        />
        <TextInput style={styles.input}
          placeholder='Setor'
          editable = {this.state.editable}
        />
        <TextInput style={styles.input}
          placeholder='Editar senha'
          editable = {this.state.editable}
        />
        <AGRButton
          text ="Editar perfil"
          style={styles.button}
          onPress={() => {this.tornarVisivel()}}
        />

      </View>
    );
  }
}


export default EditScreen;
