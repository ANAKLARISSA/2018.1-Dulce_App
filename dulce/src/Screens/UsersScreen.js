import React from 'react';
import {
  View,
  Text,
  TextInput,
  Button,
  Platform,
  Alert,
  ScrollView,
  FlatList
} from 'react-native';
import AGRButton from '../Components/AGRButton';

const styles = {
  container: {
    flex: 1,
    flexDirection: 'column',
    padding: 15,
  },
  text: {
    fontSize: 48
  }
};

class UsersScreen extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      users: [],
      loading: true
    }
  }

  componentDidMount() {
    const url = 'https://jsonplaceholder.typicode.com/users';
    this.setState({ loading: true });
    fetch(url)
    .then((response) => {
      return response.json();
    })
    .then((json) => {
      console.log(json);
      this.setState({ users: json, loading: false })
    })
  }

  render() {
    console.log(this.state);
    return (
      <ScrollView contentContainerStyle={styles.container}>
        {
          this.state.loading ? (
            <Text style={styles.text}>Carregando...</Text>
          ) : (
            <FlatList
              data={this.state.users}
              keyExtractor={(item) => {
                return item.id.toString();
              }}
              renderItem={(data) => {
                return <AGRButton text={data.item.name}  onPress={() => this.props.navigation.navigate('users')} />
              }}
            />
          )
        }
      </ScrollView>
    );
  }
}

export default UsersScreen;
