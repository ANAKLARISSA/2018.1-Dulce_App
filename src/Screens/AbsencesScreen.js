import React from 'react';
import {
  StyleSheet,
  View,
  PixelRatio,
  TouchableOpacity,
  Image,
  ScrollView
} from 'react-native';
import ScreenHeader from '../Components/ScreenHeader';
import DateRangePicker from '../Components/DateRangePicker';
import AGRButton from '../Components/AGRButton';
import ImagePicker from 'react-native-image-picker';
import {Card, CardItem, Text, Body, Form, Textarea, Picker, Right, Left} from 'native-base';

const XDate = require('xdate');

export default class AbsencesScreen extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      documentSource: null,
      absenceOption: '',
      startDay: {
        year: new XDate().getFullYear(),
        month: new XDate().getMonth() + 1,
        day: new XDate().getDate()
      },
      endDay: {
        year: new XDate().getFullYear(),
        month: new XDate().getMonth() + 1,
        day: new XDate().getDate()
      }
    };
  }

  getDocument(response) {
    /* istanbul ignore console */
    console.log('Response = ', response);

    if (response.didCancel) {
      /* istanbul ignore console */
      console.log('User cancelled photo picker');
    }
    else if (response.error) {
      /* istanbul ignore console */
      console.log('ImagePicker Error: ', response.error);
    }
    else if (response.customButton) {
      /* istanbul ignore console */
      console.log('User tapped custom button: ', response.customButton);
    }
    else {
      let source = {uri: response.uri};

      // You can also display the image using data:
      // let source = { uri: 'data:image/jpeg;base64,' + response.data };

      this.setState({
        documentSource: source
      });
    }
  }

  selectPhotoTapped() {
    const options = {
      quality: 1.0,
      maxWidth: 500,
      maxHeight: 500,
      storageOptions: {
        skipBackup: true
      }
    };

    ImagePicker.showImagePicker(options, (response) => this.getDocument(response));
  }

  onSelectDates(startDay, endDay) {
    this.setState({
      startDay: startDay,
      endDay: endDay
    });
  }

  onSelectAbsence(value) {
    this.setState({
      absenceOption: value
    });
  }

  render() {
    return (
      <View style={{flex: 1}}>
        <ScreenHeader
          title='Abono de Faltas'
        />
        <ScrollView>
          <Text style={styles.title}>
                Selecione o período de abono
          </Text>
          <DateRangePicker
            onSuccess={(startDay, endDay) => this.onSelectDates(startDay, endDay)}
            theme={{markColor: '#5f4b8b', markTextColor: 'white'}}
          />
          <Card>
            <CardItem header bordered>
            <View style={styles.view1}>
              <Text style={styles.date}>
                {this.state.startDay.day}/{this.state.startDay.month}/
                {this.state.startDay.year} até {this.state.endDay.day}/{this.state.endDay.month}/{this.state.endDay.year}
              </Text>
            </View>
            </CardItem>

            <CardItem bordered>
              <Body style={{flexDirection: 'row'}}>
                <Left>
                  <Text>
                    Selecione o abono:
                  </Text>
                </Left>
                <Right>
                  <Picker
                    selectedValue={this.state.absenceOption}
                    style={styles.picker}
                    onValueChange={(itemValue) => this.onSelectAbsence(itemValue)}>
                    <Picker.Item label='Atestado' value='medical leave' />
                    <Picker.Item label='Maternidade' value='maternity leave' />
                    <Picker.Item label='Paternidade' value='paternity leave' />
                  </Picker>
                </Right>
              </Body>
            </CardItem>
            <CardItem bordered>
              <Body style={styles.container}>
                  <TouchableOpacity onPress={this.selectPhotoTapped.bind(this)}>
                    <View style={[styles.document, styles.documentContainer, {marginBottom: 20}]}>
                      { this.state.documentSource === null ? <Text>Selecione uma Foto</Text>
                        : <Image style={styles.document} source={this.state.documentSource} />
                      }
                    </View>
                  </TouchableOpacity>
              </Body>
            </CardItem>

            <CardItem bordered>
              <Body>
                <Form style={styles.form}>
                  <Textarea rowSpan={5} bordered placeholder='Justificativa' />
                </Form>
              </Body>
            </CardItem>

            <CardItem footer bordered>
              <AGRButton
                style={styles.button}
                onPress={() => {}}
                text='Selecionar'
              />
            </CardItem>
          </Card>
        </ScrollView>
      </View>
    );
  }

}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF'
  },
  documentContainer: {
    borderColor: '#9B9B9B',
    borderWidth: 1 / PixelRatio.get(),
    justifyContent: 'center',
    alignItems: 'center'
  },
  document: {
    borderRadius: 5,
    width: 150,
    height: 150
  },
  title: {
    margin: 5,
    alignSelf: 'center',
    fontSize: 15,
    color: '#5f4b8b',
    backgroundColor: 'white'
  },
  date: {
    color: '#5f4b8b',
    fontSize: 20
  },
  dates: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'center',
    marginRight: 5,
    marginLeft: 5,
    marginTop: 10,
    borderColor: '#5f4b8b',
    borderWidth: 2,
    borderRadius: 5
  },
  justify: {
    flex: 1.5,
    backgroundColor: '#ede8f7',
    marginLeft: 5,
    marginRight: 5,
    marginTop: 10,
    borderColor: '#5f4b8b',
    borderWidth: 2,
    borderRadius: 5
  },
  button: {
    width: 200,
    marginTop: 15,
    marginRight: 5,
    marginLeft: 10
  },
  icon: {
    fontSize: 40,
    color: '#5f4b8b'
  },
  view1: {
    flex: 1,
    flexDirection: 'column',
    alignItems: 'center'
  },
  view2: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center'
  },
  view3: {
    flex: 1,
    alignItems: 'center'
  },
  form: {
    width: 375
  },
  picker: {
    height: 50,
    width: 170
  }
});
