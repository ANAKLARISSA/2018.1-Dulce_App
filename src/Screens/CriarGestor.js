import React from 'react';
import {Text, View, TextInput, Image, Alert, ScrollView } from 'react-native';
import ValidationComponent from 'react-native-form-validator';
import AGRButton from '../Components/AGRButton';


const logo = require('../../assets/img/logo.png');

const styles = {
  logo: {
    alignSelf: 'center',
    marginBottom: 20
  },
  container: {
    flex: 1,
    flexDirection: 'column',
    padding: 15,
    backgroundColor: '#FFF',
    paddingHorizontal: 50,

  },
  input: {
    height: 36,
    borderBottomWidth: 1,
    borderBottomColor: 'grey',
    marginBottom: 10,
    width: '100%',
    alignSelf: 'center',
    marginBottom: 20,
  },
  button: {
    marginTop : 30
  },
  error:{
    color: 'red'
  }
};

export default class CriarGestor extends ValidationComponent {

  constructor(props) {
    super(props);
    this.state = {
      nome: '',
      matricula: '',
      hospital: '',
      setor: '',
      senha: '',
      confirmaSenha: '',
      error: false

    }
  }

  _onPressButton(){
    if(this.validate({
        nome: {minlength:3, required: true},
        matricula: {numbers: true, required: true},
        hospital: {required: true},
        setor: {required: true},
        senha: {minlength:4, maxlength:8, required: true},
        confirmaSenha: {minlength:4, maxlength:8, required: true}
      })){
        //Metodo para mandar para a API estará aqui
        Alert.alert(
          'nome: ' + this.state.nome +
          '\nMatricula: ' + this.state.matricula +
          '\nHospital: ' + this.state.hospital +
          '\nSetor: ' + this.state.setor +
          '\nSenha: ' + this.state.senha +
          '\nConfirmação de Senha: ' + this.state.confirmaSenha
         )
      }
  }

  render() {
    return (
      <View style={styles.container}>
        <Image source={logo}  style={styles.logo} />
        <ScrollView>
        <TextInput
          ref="nome"
          style={styles.input}
          placeholder="Nome Completo"
          underlineColorAndroid="transparent"
          onChangeText={(text) => this.setState({
            nome: text})}
          />
          {this.isFieldInError('nome') && this.getErrorsInField('nome').map(errorMessage => <Text style={styles.error}>{errorMessage}</Text>) }
          <TextInput
          ref="matricula"
          style={styles.input}
          placeholder="Matrícula"
          underlineColorAndroid="transparent"
          onChangeText={(text) => this.setState({
            matricula: text})}
        />
        {this.isFieldInError('matricula') && this.getErrorsInField('matricula').map(errorMessage => <Text style={styles.error}>{errorMessage}</Text>) }
        <TextInput
          ref="hospital"
          style={styles.input}
          placeholder="Hospital"
          underlineColorAndroid="transparent"
          onChangeText={(text) => this.setState({
            hospital: text})}
        />
        {this.isFieldInError('hospital') && this.getErrorsInField('hospital').map(errorMessage => <Text style={styles.error}>{errorMessage}</Text>) }
        <TextInput
          style={styles.input}
          placeholder="Setor"
          underlineColorAndroid="transparent"
          onChangeText={(text) => this.setState({
            setor: text})}
        />

        <TextInput
          ref="senha"
          style={styles.input}
          placeholder="Senha"
          underlineColorAndroid="transparent"
          secureTextEntry
          onChangeText={(text) => this.setState({
            senha: text})}
        />
        {this.isFieldInError('senha') && this.getErrorsInField('senha').map(errorMessage => <Text style={styles.error}>{errorMessage}</Text>) }
        <TextInput
          ref="confirmaSenha"
          style={styles.input}
          placeholder="Confirme sua senha"
          underlineColorAndroid="transparent"
          secureTextEntry
          onChangeText={(text) => this.setState({
            confirmaSenha: text})}
        />
        {this.isFieldInError('confirmaSenha') && this.getErrorsInField('confirmaSenha').map(errorMessage => <Text style={styles.error}>{errorMessage}</Text>) }
        <AGRButton
          style={styles.button}
          text= 'Testar'
          onPress={() => this._onPressButton()}
        />
        </ScrollView>
      </View>
    );
  }
}
