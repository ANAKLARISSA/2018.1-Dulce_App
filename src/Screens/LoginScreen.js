import React from 'react';
import { connect } from 'react-redux';
import {View, ScrollView, TextInput, Image, Alert} from 'react-native';
import AGRButton from '../Components/AGRButton';
import AGRInput from '../Components/AGRInput';
import BotaoTransparente from '../Components/BotaoTransparente';
import axios from 'axios';
import { actionLogin } from '../Actions/currentUser';
import { NavigationActions } from 'react-navigation';

const logo = require('../../assets/img/logo.png');

const styles = {
  container: {
    flex: 1,
    flexDirection: 'column',
    padding: 15,
    backgroundColor: '#FFF'
  },
  input: {
    height: 36,
    borderBottomWidth: 1,
    borderBottomColor: 'grey',
    marginBottom: 10,
    width: '100%',
    alignSelf: 'center',
  },
  text: {
    fontSize: 36,
    color: '#039200'

  },
  logo: {
    alignSelf: 'center',
    marginTop: '20%',
    marginBottom: 40
  }
};

export class LoginScreen extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      matricula: '',
      password: ''
    };
  }

  resetNavigation(targetRoute) {
    const resetAction = NavigationActions.reset({
      index: 0,
      actions: [
        NavigationActions.navigate({ routeName: targetRoute }),
      ],
    });
    this.props.navigation.dispatch(resetAction);
  }

  _onPressButton() {
    if (this.state.password === '' || this.state.matricula === '') {
      Alert.alert(
        'Alguns campos ainda estão vazios',
        'Para entrar preencha corretamente os campos de matrícula e senha.'
      );
    } else if (this.state.password.length < 6) {
      Alert.alert(
        'Sua senha é muito curta',
        'A senha deve ter no minimo 6 caracteres.'
      );
    } else {
      //this.login();
      const token_falso = {
        matricula: this.state.matricula,
        senha: this.state.password
      };
      this.props.setCurrentUser(token_falso);
      this.resetNavigation('initial')

    }
  }

  login(){
    axios.post('http://localhost:3000/auth/users', {
      matricula: this.state.matricula, password: this.state.password
    })
      .then((response) => {
        this.props.setCurrentUser(response.data.token, response.data.id);
        this.resetNavigation('initial') })
      .catch((error) => {
        if (error.response) {
          Alert.alert(
            'Erro 1'
          );
          console.log(error.response);
        } else if (error.request) {
          Alert.alert(
            'Erro 2'
          );
          console.log(error.request);
        } else {
          Alert.alert(
            'Erro 3'
          );
          console.log(error.messege);
        }
      })
  }
  render() {
    return (
      <View style={styles.container}>
        <ScrollView>
        <Image source={logo} style={styles.logo} />
        <AGRInput
          autoCapitalize='none'
          placeholder='Matrícula'
          onChangeText={(text) => this.setState({matricula: text})}
        />

        <AGRInput
          placeholder='Senha'
          secureTextEntry
          onChangeText={(text) => {
            return this.setState({ password: text });
          }}
        />

        <AGRButton
          onPress={this._onPressButton.bind(this)}
          text="Entrar"
        />
        </ScrollView>
      </View>
    );
  }
}



const mapStateToProps = (state) => {
  return{
    currentUser:state.currentUser
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    setCurrentUser: (api_token, api_id) => {
      const currentUser = {
        token: api_token,
        id: api_id
      };
      return dispatch(actionLogin(currentUser));
    }
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(LoginScreen);
