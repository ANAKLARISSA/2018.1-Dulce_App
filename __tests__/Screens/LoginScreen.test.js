import React from 'react';
import {LoginScreen} from '../../src/Screens/LoginScreen';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import {Alert} from 'react-native';
import expect from 'expect';

require('bezier');

Enzyme.configure({adapter: new Adapter()});

jest.mock('Alert', () => {
  return {
    alert: jest.fn()
  };
});

test('renders correctly', () => {
  const tree = renderer.create(<LoginScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('change registration file ', () => {
  const wrapper = shallow(<LoginScreen />);
  const registrationField = wrapper.find('AGRInput').at(0);
  registrationField.simulate('changeText','text');
  expect(wrapper.state('registration')).toBe('text');
});

test('change password file ', () => {
  const wrapper = shallow(<LoginScreen />);
  const passwordlField = wrapper.find('AGRInput').at(1);
  passwordlField.simulate('changeText','');
  expect(wrapper.state('password')).toBe('');
});

test('test alert empty password', () => {
  const wrapper = shallow(<LoginScreen />);
  //Registration preenchido
  const registrationField = wrapper.find('AGRInput').at(0);
  registrationField.simulate('changeText','123456');
  //Password vazio
  const passwordField = wrapper.find('AGRInput').at(1);
  passwordField.simulate('changeText','');

  wrapper.findWhere(n => n.props().text === 'Entrar').simulate('Press');
  expect(Alert.alert).toHaveBeenCalled();
});

test('test alert empty regitration', () => {
  const wrapper = shallow(<LoginScreen />);
  //Registration vazio
  const registrationField = wrapper.find('AGRInput').at(0);
  registrationField.simulate('changeText','');
  //Password preenchido
  const passwordField = wrapper.find('AGRInput').at(1);
  passwordField.simulate('changeText','123456');
  wrapper.findWhere(n => n.props().text === 'Entrar').simulate('Press');
  expect(Alert.alert).toHaveBeenCalled();
});

const flushPromises = () => new Promise(resolve => setImmediate(resolve));

it('Should call login function', async() => {
  const spy = jest.spyOn(LoginScreen.prototype, 'login');
  const wrapper = shallow(<LoginScreen />);
  wrapper.setState({registration: '123456'});
  wrapper.setState({password: '123456'});
  await flushPromises();
  wrapper.update();
  const enterButton = wrapper.find('AGRButton').at(0);
  enterButton.simulate('press');
  expect(spy).toHaveBeenCalled();
});
