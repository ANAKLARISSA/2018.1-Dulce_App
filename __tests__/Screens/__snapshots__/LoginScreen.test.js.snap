// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "height": 1334,
      },
      Object {
        "backgroundColor": "#FFF",
        "flex": 1,
        "flexDirection": "column",
        "padding": 15,
      },
    ]
  }
  virtual={undefined}
>
  <Image
    source={
      Object {
        "testUri": "../../../assets/img/logo.png",
      }
    }
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "transparent",
        "height": 140,
        "marginTop": 20,
        "resizeMode": "contain",
      }
    }
  />
  <View
    style={Object {}}
    virtual={undefined}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": 1,
          "borderColor": "#D9D5DC",
          "borderLeftWidth": 0,
          "borderRightWidth": 0,
          "borderTopWidth": 0,
          "flexDirection": "row",
          "marginLeft": 15,
          "marginTop": 15,
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        collapsable={undefined}
        style={
          Object {
            "left": 0,
            "opacity": 1,
            "paddingBottom": undefined,
            "paddingTop": undefined,
            "position": "absolute",
            "right": 0,
            "top": 18,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#575757",
                "fontSize": 17,
                "left": 0,
                "paddingRight": 5,
                "top": 6,
              },
              Object {
                "fontFamily": "Lato",
                "marginTop": 5,
              },
            ]
          }
          virtual={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontSize": 17,
                },
                Object {
                  "fontFamily": "Lato",
                  "marginTop": 5,
                },
              ]
            }
            virtual={undefined}
          >
            Email
          </Text>
        </Text>
      </View>
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        editable={true}
        nameLabel="Email"
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholderTextColor="#575757"
        style={
          Array [
            Object {
              "color": "#000",
              "flex": 1,
              "fontSize": 17,
              "height": 50,
              "lineHeight": 24,
              "marginLeft": null,
              "paddingLeft": null,
              "paddingRight": 5,
              "top": 10,
            },
            Object {
              "alignSelf": "center",
              "borderBottomColor": "grey",
              "borderBottomWidth": 1,
              "fontFamily": "Lato",
              "height": 50,
              "marginBottom": 10,
              "width": "100%",
            },
          ]
        }
        underlineColorAndroid="transparent"
        value=""
        virtual={undefined}
      />
    </View>
  </View>
  <View
    style={Object {}}
    virtual={undefined}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": 1,
          "borderColor": "#D9D5DC",
          "borderLeftWidth": 0,
          "borderRightWidth": 0,
          "borderTopWidth": 0,
          "flexDirection": "row",
          "marginLeft": 15,
          "marginTop": 15,
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        collapsable={undefined}
        style={
          Object {
            "left": 0,
            "opacity": 1,
            "paddingBottom": undefined,
            "paddingTop": undefined,
            "position": "absolute",
            "right": 0,
            "top": 18,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#575757",
                "fontSize": 17,
                "left": 0,
                "paddingRight": 5,
                "top": 6,
              },
              Object {
                "fontFamily": "Lato",
                "marginTop": 5,
              },
            ]
          }
          virtual={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontSize": 17,
                },
                Object {
                  "fontFamily": "Lato",
                  "marginTop": 5,
                },
              ]
            }
            virtual={undefined}
          >
            Senha
          </Text>
        </Text>
      </View>
      <TextInput
        allowFontScaling={true}
        editable={true}
        nameLabel="Senha"
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholderTextColor="#575757"
        secureTextEntry={true}
        style={
          Array [
            Object {
              "color": "#000",
              "flex": 1,
              "fontSize": 17,
              "height": 50,
              "lineHeight": 24,
              "marginLeft": null,
              "paddingLeft": null,
              "paddingRight": 5,
              "top": 10,
            },
            Object {
              "alignSelf": "center",
              "borderBottomColor": "grey",
              "borderBottomWidth": 1,
              "fontFamily": "Lato",
              "height": 50,
              "marginBottom": 10,
              "width": "100%",
            },
          ]
        }
        underlineColorAndroid="transparent"
        value=""
        virtual={undefined}
      />
    </View>
  </View>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentContainerStyle={
      Array [
        Object {
          "padding": undefined,
        },
        undefined,
      ]
    }
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardShouldPersistTaps="handled"
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    resetScrollToCoords={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
      }
    }
    viewIsInsideTabBar={false}
    virtual={undefined}
  >
    <View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "#5f4b8b",
            "borderBottomWidth": null,
            "borderColor": "#007aff",
            "borderLeftWidth": null,
            "borderRadius": 57,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "elevation": 2,
            "flex": 8,
            "flexDirection": "row",
            "height": 50,
            "justifyContent": "center",
            "marginBottom": 0,
            "marginTop": 30,
            "opacity": 1,
            "paddingBottom": 6,
            "paddingTop": 6,
            "width": 180,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "alignItems": "center",
              "color": "white",
              "fontFamily": "Raleway",
              "fontSize": 16,
            }
          }
        >
          Entrar
        </Text>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "height": 1334,
        }
      }
      virtual={undefined}
    >
      <RCTScrollView
        automaticallyAdjustContentInsets={false}
        contentContainerStyle={
          Array [
            Object {
              "padding": undefined,
            },
            undefined,
          ]
        }
        contentInset={
          Object {
            "bottom": 0,
          }
        }
        enableAutomaticScroll={true}
        enableResetScrollToCoords={true}
        extraHeight={75}
        extraScrollHeight={0}
        getScrollResponder={[Function]}
        handleOnScroll={[Function]}
        keyboardDismissMode="interactive"
        keyboardOpeningTime={250}
        keyboardShouldPersistTaps="handled"
        keyboardSpace={0}
        onScroll={[Function]}
        resetKeyboardSpace={[Function]}
        resetScrollToCoords={
          Object {
            "x": 0,
            "y": 0,
          }
        }
        scrollEventThrottle={1}
        scrollForExtraHeightOnAndroid={[Function]}
        scrollToEnd={[Function]}
        scrollToFocusedInput={[Function]}
        scrollToPosition={[Function]}
        showsVerticalScrollIndicator={true}
        style={
          Object {
            "backgroundColor": "transparent",
            "flex": 1,
          }
        }
        viewIsInsideTabBar={false}
        virtual={undefined}
      >
        <View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "backgroundColor": "#FFF",
                "borderBottomWidth": 1,
                "borderColor": "#007aff",
                "borderLeftWidth": 1,
                "borderRadius": 57,
                "borderRightWidth": 1,
                "borderTopWidth": 1,
                "borderWidth": 1,
                "elevation": null,
                "flexDirection": "row",
                "height": 45,
                "justifyContent": "space-between",
                "marginBottom": 0,
                "marginTop": 15,
                "opacity": 1,
                "paddingBottom": 6,
                "paddingHorizontal": 20,
                "paddingTop": 6,
                "shadowColor": null,
                "shadowOffset": null,
                "shadowOpacity": null,
                "shadowRadius": null,
                "width": 280,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Image
              source={
                Object {
                  "testUri": "../../../assets/img/google.png",
                }
              }
              style={
                Object {
                  "height": 40,
                  "resizeMode": "contain",
                }
              }
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "alignSelf": "center",
                  "fontFamily": "Raleway",
                  "fontSize": 20,
                }
              }
            >
               Entrar com Google
            </Text>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "transparent",
        "borderBottomWidth": null,
        "borderColor": "#007aff",
        "borderLeftWidth": null,
        "borderRadius": 5,
        "borderRightWidth": null,
        "borderTopWidth": null,
        "elevation": 0,
        "flexDirection": "row",
        "height": 45,
        "justifyContent": "space-between",
        "opacity": 1,
        "paddingBottom": 6,
        "paddingTop": 6,
        "shadowColor": null,
        "shadowOffset": null,
        "shadowOpacity": null,
        "shadowRadius": null,
      }
    }
    testID="register"
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#5f4b8b",
          "fontSize": 15,
        }
      }
    >
      Não tem uma conta? Cadastre-se!
    </Text>
  </View>
</View>
`;
